### DISCUSSION TITLE
Trying to get current wifi network bars with WlanAPI and IpHlpAPI

### DISCUSSION THREAD

**Discussion by HolbyFPV** [dahall/Vanara#370] [Q&A]
I'm trying to recreate the wifi icon inside the Windows task bar using this library.  

To get the current network interface I am calling GetBestInterface from IpHlpAPI and it seems to be returning the right index.  However I'm struggling on how to get the RSSI or signal strength value of the wifi network that's attached to this interface.  Using this data I could recreate the wifi icon with 1 bars, 2 bars, 3 bars, ect..  

Using WlanAPI I can call WlanGetAvailableNetworkList which returns a list that has the signal strength value for each network.  There doesn't seem to be any data to connect these two things, so obviously I'm missing something. 

Anyone have any ideas on the approach I should take?

**Comment by dahall** [comment]
Have you looked at the NetworkListManager class and other networking classes in Vanara.SystemServices? It may have what you need.
